lib/services/component_manager.ts

```diff
@@ -112,6 +112,30 @@ type ComponentState = {
   sessionKey?: string
 }
 
+class ActiveComponents {
```

Comment:
Some thoughts on this class:

- The properties should be private and readonly. You should instead expose functions for querying. So instead of `this.activeComponents.uuids.includes` you would do `this.activeComponents.has(uuid)`.
- I don't think you need two separate data structures, array and record. You just need record. There seems to be an overlap here. The `add` function would just remove the if conditional. The `remove` function would remove line 128. And the `clear` function would remove line 135. Then you'd introduce a function called `has` that checks to see if `this.areas[uuid] !== undefined`.
- Is this class really required? Sounds like you could just do the same thing with a Record property. Or maybe a Set property. It does offer some abstraction I suppose.

Reply:
I moved away from a Record because I had made an assumption that I realize is no longer true anymore. I'll walk that back.